src/EnergyPlus/ConvectionCoefficients.hh

```diff
@@ -146,38 +146,51 @@ namespace ConvectionCoefficients {
     int constexpr HcExt_AlamdariHammondUnstableHorizontal{325};
 
     // parameters, by zone, for flow regimes for adaptive convection on inside face
-    int constexpr InConvFlowRegime_A1{1}; // In-floor heating or in-ceiling cooling
-    int constexpr InConvFlowRegime_A2{2}; // In-wall heating
-    int constexpr InConvFlowRegime_A3{3}; // no HVAC system, all bouyancy
-    int constexpr InConvFlowRegime_B{4};  // Convective heater in zone
-    int constexpr InConvFlowRegime_C{5};  // central mechanical air
-    int constexpr InConvFlowRegime_D{6};  // zone mechanical air
-    int constexpr InConvFlowRegime_E{7};  // mixed. mechancial air and bouyancy
+    enum class InConvFlowRegime
+    {
+        Invalid,
```

Comment:
Set `Invalid` to -1 and have the valid series start at 0. That is the pattern we are using with `enum`.

Reply:
I’m curious if this is more about consistency or if there’s something later that will require this pattern? This particular one doesn’t need an int representation anywhere.

I’ll change it either way, just curious.

Reply:
Yes. We are going to change the way in which `enum`s are mapped to `string`s (for display) and vice versa (for construction) and the latter will use -1 (which is the index you get when something is not found in a list) to indicate that the given `string` is not a valid option for that `enum`.

Reply:
Understood. Thanks for the clarification.

Reply:
FWIW this new idiom will also be our first use of `gsl::span`.